src/main/java/htsjdk/io/HtsPath.java

```diff
@@ -282,19 +283,36 @@ private URI getURIForString(final String pathString) {
     }
 
     /**
-     * check that there isn't a non file scheme at the start of the path
-     * @param pathString
-     * @param cause
+     * Check for problems associated with the presence of a heirachical scheme.
+     *
+     * It's better to reject cases like `://` or `ftp://I forgot to encode this` than to treat them as relative file paths
+     * It's almost certainly an error on the users part instead of an atttempt to intentionally reference a file named
+     * `file:///workingidr/ftp:/I forgot to encode this`
+     *
+     * Note this is only meant to be called in the case of a URLSyntaxException already having occured during initial
+     * parsing of the URI
+     *
+     * @param pathString the path being examined
+     * @param cause the original failure reason
      */
-    private static void assertNoNonFileScheme(String pathString, URISyntaxException cause){
-        final String[] split = pathString.split(":");
+     static void assertNoProblematicScheme(String pathString, URISyntaxException cause){
+        if(pathString.equals(HIERARCHICAL_SCHEME_SEPARATOR)){
+            throw new IllegalArgumentException(HIERARCHICAL_SCHEME_SEPARATOR + " is not a valid path.", cause);
+        }
+
+        if(pathString.endsWith(HIERARCHICAL_SCHEME_SEPARATOR)){
+            throw new IllegalArgumentException("A path consisting of only a scheme is not allowed: " + pathString, cause);
```

Comment:
Hmm...the string `http://host/path://` also ends with the HIERARCHICAL_SCHEME_SEPARATOR, but has a scheme?

Reply:
thats.... a good point